### ISSUE TITLE
Do I need Enterprise for AWS at-rest/in-transit encryption?

### ISSUE THREAD

**Issue by kceb** [mperham/sidekiq#3755]
More so a question than an issue.
[I noticed Sidekiq enterprise supports encryption](https://github.com/mperham/sidekiq/wiki/Ent-Encryption). I'm looking to use [at-rest and in-transit encryption](https://docs.aws.amazon.com/AmazonElastiCache/latest/UserGuide/in-transit-encryption.html) via an aws redis elasticache instance. I'm wondering if Sidekiq enterprise is necessary for this. I tried connecting normally to an encrypted instance and sidekiq wouldn't start. When I disabled the encryption, it worked. This leads me to believe there's a config option I'm missing.

Ruby version: 
* ruby 2.3.4

Sidekiq / Pro / Enterprise version(s): 
* Sidekiq Free

Please include your initializer and any error message with the full backtrace.
* Not relevant

Are you using an old version?
* Yes

Have you checked the changelogs to see if your issue has been fixed in a later version?

https://github.com/mperham/sidekiq/blob/master/Changes.md
https://github.com/mperham/sidekiq/blob/master/Pro-Changes.md
https://github.com/mperham/sidekiq/blob/master/Ent-Changes.md

**Comment by mperham** [comment]
There's many different points and layers at which you can enable encryption.

* TLS can be used to encrypt job data in flight over the network.
* AWS can encrypt all network traffic between machines in your internal network
* Sidekiq Enterprise will encrypt the data when it is outside of the Sidekiq process (both in-flight and at-rest in Redis).  It's the highest-level encryption so it will provide the most coverage; TLS won't provide encryption of job data within the Web UI, for instance.

In other words, you need to decide what level you want to encrypt at.
